standard/currency/currency.lua

```diff
@@ -92,4 +92,18 @@ function Currency.formatMoney(value)
 	return LANG:formatNum(integer) .. string.format('%.2f', decimal):sub(2)
 end
 
+function Currency.getExchangeRate(props)
+	local setVariables = Logic.readBool(props.setVariables)
+	local currencyRate = tonumber(props.currencyRate)
+	if not currencyRate then
+		currencyRate = mw.ext.CurrencyExchange.currencyexchange(1, props.currency:upper(), USD, props.date)
```

Comment:
Note: This function can return `false`.

Reply:
in which case?

if you are referring to
> In case of an error, the parser function will provide an explicit error message. The Lua version will instead return false.

that seems outdated:
https://liquipedia.net/commons/Module:HjpSandbox
![DFDB18F3-D426-48CC-A94D-43235DE74721](https://user-images.githubusercontent.com/75081997/183320608-5b2c2674-39f2-47bf-b65b-bd5c3a8f54d8.jpeg)

from the looks of it either the exchange extension errors with an explicit error message or it returns a value or nil

Reply:
I had some tests with it a couple of weeks ago and it returned false. I think it was for currency that didn't exist.

Reply:
Ah, the `tonumber()` in the return will cause the `false` to become `nil`.